src/NUnit.Portable.Agent/NUnitPortableDriver.cs

```diff
@@ -69,17 +65,15 @@ public class NUnitPortableDriver
         public string Load(Assembly frameworkAssembly, Assembly testAssembly, IDictionary<string, object> settings)
         {
             var idPrefix = string.IsNullOrEmpty(ID) ? "" : ID + "-";
-            _frameworkAssembly = frameworkAssembly;
-            _testAssembly = testAssembly;
-
-            _frameworkController = CreateObject(CONTROLLER_TYPE, testAssembly, idPrefix, settings);
-            if (_frameworkController == null)
+            var frameworkController = CreateObject(CONTROLLER_TYPE, frameworkAssembly, testAssembly, idPrefix, settings);
+            if (frameworkController == null)
                 throw new NUnitPortableDriverException(INVALID_FRAMEWORK_MESSAGE);
 
-            _frameworkControllerType = _frameworkController.GetType();
+            var testPackage = new TestPackage(testAssembly, frameworkAssembly, frameworkController);
+            _testPackages.Add(testPackage);
 
-            log.Info("Loading {0} - see separate log file", _testAssembly.FullName);
-            return ExecuteMethod(LOAD_METHOD) as string;
+            log.Info("Loading {0} - see separate log file", testAssembly.FullName);
+            return testPackage.ExecuteMethod(LOAD_METHOD) as string;
         }
```

Comment:
In the framework, load returns only the top level Xml node for the assembly. That's because the result may be remoted and not all runners need more than the top level. Runners that need more call Explore, which gets the entire content. In the console, that load result is used because it may contain error messages. I think you need to do that as soon as you start loading multiple assemblies unless you want the entire run to fail when one assembly cannot be found.